src/inference/graph_inference.cpp

```diff
@@ -341,6 +342,24 @@ class GraphNodeAssignment : public Nice2Assignment {
     return label_set_->GetLabelName(label_id);
   }
 
+  // Returns conditional probability of a label conditionated on its neighbouring labels
```

Comment:
I see no benefit of having these two methods. The formula would be far more understandable if you just include the calculations in PLLearn.